### ISSUE TITLE
Unable to use dsaas-stg cluster as target for tenant namespaces

### ISSUE THREAD

**Issue by alexeykazakov** [openshiftio/openshift.io#4574]
As part of our OSD support effort I'm experimenting with targeting our dsaas-stg OSD cluster to provision all tenant namespaces for a test user.

It's still a single cluster environment I just replaced the OSO starter cluster by our dsaas-stg cluster for one test user I specially created for that.

I faced bunch of issue which we should address.

- [ ] Can't open jenkins via OSIO UI / jenkins idler or approve the prod rollout in the pipeline view. I guess I'm facing the workaround which was introduced to fix https://github.com/fabric8-services/fabric8-jenkins-idler/issues/272 and which filters out all not OSO starter clusters - https://github.com/fabric8-services/fabric8-jenkins-idler/blob/33d06182dfc6ea408df16ef35703776b2c5251ee/internal/cluster/service.go#L23-L27 -- [Idler issue #304](https://github.com/fabric8-services/fabric8-jenkins-idler/issues/304)
Webhooks from GitHub also do not work because of the same issue.
```
{"Errors":[{"code":"500","detail":"could not find entry for cluster https://api.dsaas-stg.openshift.com/"}]}
```
- [ ] Stage/Run deployments fail:
![deploy](https://user-images.githubusercontent.com/620087/48881425-4913a100-edca-11e8-8654-4979794120cc.png)
- [x] Can't start Che workspace:
![che](https://user-images.githubusercontent.com/620087/48881436-5af54400-edca-11e8-95d1-353f2d10dce9.png)

**Comment by alexeykazakov** [comment]
@ibuziuk @chmouel @sthaha could you guys take a look? It's quite important because it's for our OSD support.

**Comment by ibuziuk** [comment]
@alexeykazakov could you please provide events from the *-che namespace ?

**Comment by chmouel** [comment]
@alexeykazakov idler should not be enabled on OSD, isnt it ? as for webhook we are buidling a new service for the webhook handling which would take care of OSD not having the webhook going to proxy, as for  fabric8-jenkins-proxy it needs to be modified to take care of OSD,

**Comment by alexeykazakov** [comment]
@chmouel correct. We don't need idler for OSD. 
But let me clarify a bit what I'm doing. My intention is to test what we have now and collect possible issues unexpected issues as early as possible. By using this test account we can start testing things earlier without need to wait for full OSD support. If there are some issues like GitHub hooks which we are already aware of and it's being addressed and which we can workaround right now (by triggering the build manually for example) then it's fine. But if there are some issues which blocks us from further testing then it would be nice to fix them right now even if it's a quick and short term workaround until proper long-term solution is in place. This would unblock our testing.
For example I can't deploy anything to stage/run so it's hard to say if we have any additional issues there which we do not expect.
Does it make sense?

**Comment by chmouel** [comment]
@alexeykazakov fair enough so if I understand properly you want a quick workaround before +proxy and +webhook implementation are done which is a fair request to ask since this is going to take a while, 

Is there any chances we can get a `OSD`  flag in fabric8-cluster ? that would help a lot, we would just need to get proxy to respect it and go directly to the jenkins without checking via idler if he sees it,

**Comment by alexeykazakov** [comment]
@ibuziuk first at all when I tried to re-open the workspace in OSIO UI (codebases) the che in my namespace didn't unidle I had to unidle it manually in the OpenShift console.
The workspace deployment failed with the following (only) event:
`Error creating: pods "workspacefvhca2jjqttcph8d.dockerimage-75589c545-" is forbidden: error looking up service account alexey-osio-dsaas-stg-che/che-workspace: serviceaccount "che-workspace" not found`

**Comment by alexeykazakov** [comment]
@chmouel the webhook is not a big deal right now because we can workaround it (manually start a new build) but still would be nice to see it working so we can make sure we don't have other issues with hooks.. But it's not our top priority for sure :)
Proxy is more important. It blocks quite a few important functionality. Why deployments fail? Is it related to proxy somehow?

Regarding the OSD/OSO flag it's fair request. I've created an issue for that - https://github.com/fabric8-services/fabric8-cluster/issues/31
We will try to provide it ASAP. cc: @dipak-pawar

**Comment by chmouel** [comment]
@alexeykazakov so after looking at the code properly : 

proxy get clusters url from idler
idler is filtering those clusters that are not oso (with regexp) 
proxy fail with a `could not find entry`

what we want is getting the list from cluster on idler still but only have idler active on OSO.

We can continue do this with regexp but would be nicer to do with a flag,

I have created an issue here : 

https://github.com/fabric8-services/fabric8-jenkins-idler/issues/304

**Comment by ibuziuk** [comment]
> @ibuziuk first at all when I tried to re-open the workspace in OSIO UI (codebases) the che in my namespace didn't unidle I had to unidle it manually in the OpenShift console.
> The workspace deployment failed with the following (only) event:
> Error creating: pods "workspacefvhca2jjqttcph8d.dockerimage-75589c545-" is forbidden: error looking up service account alexey-osio-dsaas-stg-che/che-workspace: serviceaccount "che-workspace" not found

@alexeykazakov looks like you do not have latest version of che template for some reason - `che-workspace` sa [1] should have been created by in your *-che namespace by init-tenant.

[1] https://github.com/fabric8-services/fabric8-tenant/blob/master/environment/templates/fabric8-tenant-che-mt.yml#L56

**Comment by alexeykazakov** [comment]
@MatousJobanek it looks like that tenant service provisioned the single tenant che template instead of the default multi tenant che template. Could you please investigate why?

**Comment by alexeykazakov** [comment]
@MatousJobanek we discussed it with @ibuziuk and Ilya said that we can just remove the single tenant templates completely. It's not used.

**Comment by MatousJobanek** [comment]
That is great news, I'll be more than happy to remove it. I'll create an issue as well as a PR in the tenant project and link it here.

**Comment by dipak-pawar** [comment]
> @chmouel the webhook is not a big deal right now because we can workaround it (manually start a new build) but still would be nice to see it working so we can make sure we don't have other issues with hooks.. But it's not our top priority for sure :)
Proxy is more important. It blocks quite a few important functionality. Why deployments fail? Is it related to proxy somehow?
Regarding the OSD/OSO flag it's fair request. I've created an issue for that - fabric8-services/fabric8-cluster#31
We will try to provide it ASAP. cc: @dipak-pawar

@chmouel fabric8-services/fabric8-cluster#31 is merged to master and ready to use in prod-preview which means you have cluster type availble in `api/clusters` api.

cc @alexeykazakov

**Comment by chmouel** [comment]
@dipak-pawar thanks, i have added this https://github.com/fabric8-services/fabric8-jenkins-idler/issues/306

**Comment by MatousJobanek** [comment]
> @MatousJobanek we discussed it with @ibuziuk and Ilya said that we can just remove the single tenant templates completely. It's not used.

@alexeykazakov @ibuziuk here is PR that removes it: https://github.com/fabric8-services/fabric8-tenant/pull/690

**Comment by alexeykazakov** [comment]
Che works fine with updated template:

![osd-che](https://user-images.githubusercontent.com/620087/50194163-80546f80-02ee-11e9-9f0c-df3191420cd2.png)

**Comment by ibuziuk** [comment]
@alexeykazakov should this issue be closed as outdated / won't fix ?

**Comment by alexeykazakov** [comment]
I agree. Closing as outdated.
